akka-actor-testkit-typed/src/test/java/akka/actor/testkit/typed/javadsl/BehaviorTestKitTest.java

```diff
@@ -347,23 +347,22 @@ public void recordWatchWith() {
   @Test
   public void allowRetrievingAndKilling() {
     BehaviorTestKit<Command> test = BehaviorTestKit.create(behavior);
-    TestInbox<ActorRef<String>> i = TestInbox.create();
     TestInbox<String> h = TestInbox.create();
-    test.run(new SpawnSession(i.getRef(), h.getRef()));
 
-    ActorRef<String> sessionRef = i.receiveMessage();
-    assertFalse(i.hasMessages());
+    ReplyInbox<ActorRef<String>> sessionReply =
+        test.ask(replyTo -> new SpawnSession(replyTo, h.getRef()));
+
+    ActorRef<String> sessionRef = sessionReply.receiveReply();
+
     Effect.SpawnedAnonymous s = test.expectEffectClass(Effect.SpawnedAnonymous.class);
     assertEquals(sessionRef, s.ref());
 
     BehaviorTestKit<String> session = test.childTestKit(sessionRef);
     session.run("hello");
     assertEquals(Collections.singletonList("hello"), h.getAllReceived());
 
-    TestInbox<Done> d = TestInbox.create();
-    test.run(new KillSession(sessionRef, d.getRef()));
+    test.<Done>ask(replyTo -> new KillSession(sessionRef, replyTo)).expectReply(Done.getInstance());
```

Comment:
Not the best ergonomics for inference here in Java... a variant (a la `javadsl.ActorContext.ask`) which takes the response class would allow:

```java
test.ask(Done.class, replyTo -> new KillSession(sessionRef, replyTo)).expectReply(Done.getInstance);
```

Reply:
I agree that the `.<Done>ask` syntax is typically not well known.
Assigning the result is often good, anyway.
However, there could be an advantage of aligning with `javadsl.ActorContext.ask`, which has a Class parameter.

Reply:
I don't think the class-taking version should replace the current version, as the overall situation (being outside a behavior) is more like `typed.ActorRef.ask`, where the `ReplyInbox` takes the place of the `CompletionStage`.  The inline style, though, is much more like `javadsl.ActorContext.ask`.